### ISSUE TITLE
Error execution

### ISSUE THREAD

**Issue by tXambe** [smicallef/spiderfoot#1635]
Please provide a description of the issue and any relevant error messages.
Hello when execute the s`f.py I have this error:

```
sudo python3 sf.py -l 81.45.14.153:443                                                                             

*************************************************************
 Use SpiderFoot by starting your web browser of choice and 
 browse to http://81.45.14.153:443/
*************************************************************

2022-02-18 09:45:20,507 [INFO] sf : Starting web server at 81.45.14.153:443 ...
2022-02-18 09:45:20,517 [WARNING] sf : 
********************************************************************
Warning: passwd file contains no passwords. Authentication disabled.
Please consider adding authentication to protect this instance!
Refer to https://www.spiderfoot.net/documentation/#security.
********************************************************************

/usr/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 3 leaked semaphore objects to clean up at shutdown                                   
  warnings.warn('resource_tracker: There appear to be %d '
```

If you can provide an application stack trace that is even better.

What version of Python are you using?

Python 3.9.10

What version of SpiderFoot are you using (stable release or Git `master` branch)?

SpiderFoot 3.5.0: Open Source Intelligence Automation.

You may also wish to check if your issue has been posted previously:

* https://github.com/smicallef/spiderfoot/issues

A greeting and thanks

**Comment by bcoles** [comment]
Does SpiderFoot work, regardless of the warning?

**Comment by rdef** [comment]
I also have this issue.


What version of Python are you using?

- Python 3.10.0 on 2021 M1 MacBook

What version of SpiderFoot are you using (stable release or Git master branch)?

- SpiderFoot 3.5.0


```
[user]@192-168-1-103 spiderfoot-3.5 % python ./sf.py -l 127.0.0:5001
2022-03-29 15:49:23,184 [INFO] sf : Starting web server at 127.0.0:5001 ...

*************************************************************
 Use SpiderFoot by starting your web browser of choice and 
 browse to http://127.0.0:5001/
*************************************************************
2022-03-29 15:49:23,191 [WARNING] sf : 
********************************************************************
Warning: passwd file contains no passwords. Authentication disabled.
Please consider adding authentication to protect this instance!
Refer to https://www.spiderfoot.net/documentation/#security.
********************************************************************


[user]@192-168-1-103 spiderfoot-3.5 % /Users/[user]/.pyenv/versions/3.10.0/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 3 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

**Comment by smicallef** [comment]
This issue appears to be caused by something else occupying the port you've chosen to use for SpiderFoot. Check `netstat` to see what else might be listening on port 5001.

**Comment by geeknik** [comment]
I only see this error message when running `python3 sf.py -C scan_ID` on a large database. For example, I have one scan with 151619 elements, and after a few minutes of running the correlations, this error message appears. Doesn't seem to have an affect on anything that I can tell.

**Comment by smicallef** [comment]
Just curious, when you run that, is another `sf.py` already running on the machine?

**Comment by geeknik** [comment]
No. In fact here is what I see from a run against the 151619 element scan:
```
spiderfoot@dev:~/spiderfoot$ python3 sf.py -C 1234567
2022-04-19 18:12:37,724 [INFO] sf : Running 38 correlation rules against scan, 1234567
Terminated
spiderfoot@dev:~/spiderfoot$/usr/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 3 leaked semaphore objects to clean up at shutdown.
 warnings.warn('resource_tracker: There appear to be %d'
 ```
 
 And then it returns to a command prompt when I hit enter.. This machine has 16GB of RAM allocated but I think there might be a `ulimit` issue happening here as I can't get the correlation run to continue.. The original scan stalled out too and didn't make any requests for like 24 hours according to the log.. No other error messages seen..

**Comment by bcoles** [comment]
> I'm having the same problem is there a fix yet?

The above issue description is unclear. What error are you encountering exactly?

Someone encountered a warning message when starting SpiderFoot. Someone encountered an error when running correlations on an existing large scan. Someone (on discord) encountered the warning when running a scan.

The warning is a warning. Usually SpiderFoot will continue to operate fine.

One possible issue is an inadequate `ulimit` for large scans. The output above which shows `Terminated` was likely due to insufficient `ulimit` or insufficient memory. Try increasing the `ulimit`, in particular the number of maximum open files (`ulimit -n`). Perhaps also stack limit and virtual memory.

**Comment by bcoles** [comment]
> spiderfoot -l IP:PORT

Presumably you are using the `spiderfoot` operating system package. This isn't something I can easily test at the moment.

Depending how it is configured you may be able to get some error information from the log files. SpiderFoot writes log files to `$HOME/.spiderfoot/logs` but the Kali system service may put them somewhere else (`/var/log`?). You may also be able to find some logs with `journalctl -u spiderfoot`. if something catastrophic has occurred you may find info in `dmesg`. You may also want to check that nothing else is listening on port 5001 already.

**Comment by bcoles** [comment]
> spiderfoot.error.log -
> 2022-07-31 05:58:46,818 [CRITICAL] sf.py:241 : Invalid ip:port format.
> 2022-07-31 06:00:47,196 [WARNING] sf.py:550 :

This implies you have entered an incorrect ip:port format.

Ensure you enter a correct ip:port format.

**Comment by bcoles** [comment]
I created a new kali 2022.02 VM and installed SpiderFoot with `apt-get install spiderfoot`.

The web interface worked out of the box. I can't reproduce this issue.

![SpiderFoot](https://user-images.githubusercontent.com/434827/182027300-53f0c238-7ded-408f-88a6-f4d5b24e4d99.png)

**Comment by bcoles** [comment]
I was able to reproduce this issue by starting something else on port 5001 first.

![image](https://user-images.githubusercontent.com/434827/182027555-4c527756-235e-4bf7-858e-eac58bab518b.png)

I thought SpiderFoot contained a check for this, but apparently not. You can check if something is already listening on port 5001 with `lsof -i :5001`.

**Comment by bcoles** [comment]
> how do i check which port is empty

If by "empty" ports you mean available ports, that would simply be the inverse of the open ports (maximum of 65,535).

**Comment by bcoles** [comment]
> yeah but how do i check what ports are availabel

There are 65,535 TCP ports and 65,535 UDP ports. All ports which are not in use are available. However, you will need elevated privileges to bind to privileged ports (ports below 1024).

This is unrelated to SpiderFoot.

**Comment by bcoles** [comment]
> so what should i do to solve this error? what command or fix do i need to do

Please show the terminal output for the following command:

```
spiderfoot -l 127.0.0.1:12345
```
